packages/block-library/src/gallery/index.php

```diff
@@ -130,7 +116,47 @@ function block_core_gallery_render( $attributes, $content ) {
 			'context' => 'block-supports',
 		)
 	);
-	return (string) $processed_content;
+
+	$updated_content = $processed_content->get_updated_html();
```

Comment:
`$processed_content` should be a `WP_HTML_Tag_Processor` object. However, even without converting to a string on this line, the rest of the processing seems to run correctly and the blocks also are rendered correctly 🤔

Reply:
Good catch. It's because the class [implements](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/html-api/class-wp-html-tag-processor.php#L3114) the [`__toString`](https://www.php.net/manual/en/language.oop5.magic.php#object.tostring) function which when the instance is treated as a string just returns the `get_updated_html()` value. 

I think it is better to be explicit here to reduce the potential for the confusion you just highlighted.

Reply:
I've updated it by 323cdec, but if I haven't explained it well, feel free to update it 🙇